Constellation can now optionally report memory usage when the page is loaded.

Narfinger opened the pull request:
The constellation can now tell the memory reporter to report the memory to a trace file when a page is loaded.
Additionally, we amend the memory reporter to allow a simple message where it will report the memory to a tracing provider (at the moment only OHOS/hitrace is supported but easy extension is possible).

I am not sure if this is the right approach or if the embedder should decide to have the memory reporting done.

Testing: This does not change functionality of any of the rendering.

jschwe commented:
Maybe it would be nicer to request the report from the embedder, and then dump the reports from servoshell. It's not like the reports are timing sensitive. 
I guess we could also consider dumping the report to a file, if that makes parsing easier, or allows us to log information more flexibly.

Narfinger committed:
Allow the embedder to request memory reporting snapshot.

The embedder can now tell the memory reporter to report the memory back to it. We also report the memory on LoadPage::Complete if hitrace is enabled.

Testing: This does not change functionality of any of the rendering.

Signed-off-by: Narfinger <Narfinger@users.noreply.github.com>

Narfinger commented:
Refactored the code according to suggestions.

jdm reviewed (approved):
Nice!

Narfinger committed:
incorporate suggestions

Signed-off-by: Narfinger <Narfinger@users.noreply.github.com>
